packages/react-core/src/components/Select/examples/SelectTypeahead.tsx

```diff
@@ -44,8 +46,9 @@ export const SelectBasic: React.FunctionComponent = () => {
       // When no options are found after filtering, display 'No results found'
       if (!newSelectOptions.length) {
         newSelectOptions = [
-          { isDisabled: false, children: `No results found for "${filterValue}"`, value: 'no results' }
+          { isAriaDisabled: true, children: `No results found for "${filterValue}"`, value: NO_RESULTS }
         ];
+        resetActiveAndFocusedItem();
```

Comment:
This isn't a blocker and more of a nit, but it is odd what you had mentioned about the focus styling for the "no results" option.

We could replace this `reset...` call and instead `setActiveAndFocusedItem` to keep focus on the first item regardless. Or we could maybe try to combine the 2 useEffects into a single one to better ensure focus never gets placed on a "no results" option. 

Another option is to just not auto place the "focus" on the first item, which is something that a [W3C combobox example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/) is doing. Which that actually makes sense when testing with VoiceOver. As I type into the input, I may not get any indication that the first item is "focused", so if I press down arrow to traverse the list it sounds like I'm "starting" on the second item. Sometimes I can hear VO saying "[first item] not selected...", but sometimes it may not do that or it may get cutoff as it announces what I've typed into the input. Preventing the "autofocus" would mean pressing an arrow key brings me to the expected first (or last, if Up arrow is pressed) item. Curious what others think, though.

Reply:
I really like the other option you are mentioning about not auto focusing the first option. As you say, it makes sense with the voice over, and it would also solve the unwanted focusing of the "no results" option.

If we would go with this option, we should then prevent selecting the first option on pressing "enter" - it should only open/close the menu.

About updating other examples - I agree with that, I'll update them once we agree on the final version of the  Typeahead behaviour.

Reply:
Eric and I were talking about this earlier, and I agree with both of you that it would probably make sense to not auto focus the first option and just shift "focus" to the first option when the user presses the arrow keys (which I believe is what the W3C combobox example is doing). This would probably help improve the screen reader experience.